### PR TITLE
Allocate less in GetIPEndPoint

### PR DESCRIPTION
**Pull request by pakrym**
Trim 2 byte arrays creation for IPv6 addresses.

### CODE REVIEW

**Comment by pakrym** [comment]
cc @davidfowl @halter73 @cesarbs

**Comment by pakrym** [comment]
This happens per connection, so might be not have hight impact.

**Comment by halter73** [comment]
:shipit:

File: src/Microsoft.AspNetCore.Server.Kestrel/Internal/Networking/SockAddr.cs [hunk 1]
```diff
@@ -84,16 +84,11 @@ public IPEndPoint GetIPEndPoint()
...
                 // otherwise IPv6
-                var bytes1 = BitConverter.GetBytes(_field1);
-                var bytes2 = BitConverter.GetBytes(_field2);
-
                 var bytes = new byte[16];
-                for (int i = 0; i < 8; ++i)
-                {
-                    bytes[i] = bytes1[i];
-                    bytes[i + 8] = bytes2[i];
+                fixed (byte* b = bytes) {
```
Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Networking/SockAddr.cs]: nit: `{` in new line

File: src/Microsoft.AspNetCore.Server.Kestrel/Internal/Networking/SockAddr.cs [hunk 2]
```diff
@@ -84,16 +84,11 @@ public IPEndPoint GetIPEndPoint()
...
+                    *((long*)b) = _field1;
+                    *((long*)(b + 8)) = _field2;
```
Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Networking/SockAddr.cs]: nit: don't remove blank line after block

**Comment by cesarbs** [comment]
:ship:

**Pull request merged**